### DISCUSSION TITLE
How long will it take for solid-start to reach 1.0?

### DISCUSSION THREAD

**Discussion by jqhr** [solidjs/solid-start#1102] [Q&A]
We have waited too long for 1.0

**Comment by atk** [comment]
Solid has the advantage of being a pure open source project without commercial interests. That allows the developers to take their time to find the best possible solutions instead of making forced compromises just to be faster. So it won't be prematurely presented as 1.0 to the public. "Ready when done" applies.

**Reply by jqhr** [reply]
Okay

**Comment by rolanday** [comment]
> "Solid has the advantage of being a pure open source project without commercial interests. That allows the developers to take their time to find the best possible solutions instead of making forced compromises just to be faster. So it won't be prematurely presented as 1.0 to the public. "Ready when done" applies."

I disagree with this mindset (^^).

I super appreciate the amazing the work and thought leadership coming out of Sold team (and put my money where my mouth is by backing). Still, it's a fair question because calling something Beta suggests nearing completion and I think devs spending countless hours helping to support the project through adoption, filing bugs, etc., deserve something more than "it will be done when it's done". Also, appreciate that Start is needing to take a reset w/ Beta 2, but would at least like to get some idea when we can expect it to be available. I've paused all Solid work in the meantime.

**Reply by atk** [reply]
Thanks for elaborating on your position and thank you for sponsoring this project.

I really believe the current second beta will be the last one before 1.0, but I do not see a fixed eta for 1.0.

I too am one of the devs who tested it a few times, also filed and even fixed a few issues in the previous beta. For the current development, I'm more of a spectator, though.

My answer does not imply that the question is unfair, just that I see no fixed timeline, but a road map, and I really appreciate that, because I've seen too many projects fail due to hasty compromises.

**Comment by rolanday** [comment]
thx atk. I just think it would behoove Solid leadership team to set expectations on timelines -- if adoption is a concern, and, development community deserves it too because most also working for free, non-commercial, w/ time invested to help Solid succeed. Or, if Solid is simply being treated as a science project to prove befits of signals for use in other frameworks, that's ok too, but should be clear about expectations IMO. Still, looking forward to Beta 2! so I can re-engage w/ Solid :-)

**Comment by jqhr** [comment]
I think solid-start should have a clear roadmap, and I also want to participate in the solid-start project so that 1.0 can come quickly.

**Comment by ryansolid** [comment]
It is nearly impossible to give a clear timeline for 1.0 at this point because we are working on conditions that were only the people currently working on SolidStart able to do so we'd be several months out. It's basically trying to estimate something when the work overshadows the amount of resources and the consistency of those to such a degree that any variable could drastically shift things. Instead trying to focus at a couple goals at a time. With the understanding that as the ball gets rolling things swing in our favor.

I understand this is the perpetual chicken in the egg problem for those waiting on 1.0. But it is a reality I've had to come to face in the 7 years I've been doing this. Good work takes time and resources, which beget more time and resources. We continue to grow and our ability to do more increases but so does our ambition. Without timelines people can't make commitments for adoption and that is something that we've had to come to terms with over time, because the last thing I would ever do is lie to you.

It isn't that we're doing some experiment here, but that we are dealing with 3-30x less resources than most of what you'd consider our nearest competitors. So we do what we can. In the Solid Beta 2 discussion I've outlined the work that has to be done just not when. Some of it concrete, some of it requiring research. That research is something I'm prioritizing personally over even the work on Solid 2.0. But there is no point trying to ship something when we know it is going to change almost immediately.

Thank you for your patience, and I recognize this impacts our ability for adoption, but it is much better than the alternative. Right now we have the ability to set us up for success and we can't squander that by being hasty. We are at the a turning point in frontend frameworks of the like we haven't seen in over a decade and it is critically important that we hit this wave in stride. When the next few things are resolved and start the second Beta there will be opportunity again to help out which should accelerate our plans.
